doc/architecture-decisions/0012-routecontroller-route-deletion-finalizer.md

```diff
@@ -0,0 +1,27 @@
+# 11. Routecontroller Route Deletion Finalizer
+
+Date: 2020-05-11
+
+## Status
+
+Accepted
+
+## Context
+Without a finalizer, we could rely on cascading deletion of child objects when Routes are deleted and `RequeueAfter` to rebuild the virtual services and services  from its routes every `ResyncInterval` seconds. `ResyncInterval` is currently set to 30 seconds.
```

Comment:
Can you link to the Kubernetes/Kubebuilder docs on finalizers somewhere in this section?

* https://book.kubebuilder.io/reference/using-finalizers.html
* https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers

Reply:
These links are now in the paragraph above this one :)